pkgs/development/python-modules/dagger-io/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, anyio
+, attrs
+, beartype
+, buildPythonPackage
+, cattrs
+, colorama
+, fetchFromGitHub
+, gql
+, graphql-core
+, httpx
+, platformdirs
+, poetry-core
+, pythonOlder
+, rich
+, shellingham
+, strawberry-graphql
+, typer
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "dagger-io";
+  version = "0.6.1";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.10";
+
+  src = fetchFromGitHub {
+      owner = "dagger";
+      repo = "dagger";
+      rev = "v${version}";
+      sha256 = "sha256-9QQ6aDCkTWNq5KOSGF6FH6UQrOYa51ctW3CMcGrCJAQ=";
+  } + "/sdk/python";
+
+  propagatedBuildInputs = [
+    anyio
+    attrs
+    beartype
+    cattrs
+    colorama
+    gql
+    graphql-core
+    httpx
+    platformdirs
+    poetry-core
```

Comment:
You are sure poetry-core is not only used to build the package? Then it should go to nativeBuildInputs

Reply:
It doesn't build without so I think it's fine to leave it.

Reply:
`poetry-core` should go to `nativeBuildInputs` since it is a build-time only dependency.

Reply:
I'll try it tomorrow and change it to if it works.